server/api_archive_integration_test.go

```diff
@@ -0,0 +1,255 @@
+package server
+
+import (
+	"archive/zip"
+	"bytes"
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"os"
+	"path/filepath"
+	"slices"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/glowlabs-org/gca-backend/glow"
+)
+
+// This test creates a running server, populated with some report data, then
+// gets an archive and compares to the original file data.
+func TestApiArchive(t *testing.T) {
+	// Create a populated test environment and start a new server.
+	gcas, dir, err := ServerWithArchiveFiles(t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer gcas.Close()
+
+	fileMap := map[string]bool{}
+	dataMap := map[string][]byte{}
+
+	// Add the public files to the test.
+	for _, name := range PublicFiles {
+		path := filepath.Join(dir, name)
+		data, err := os.ReadFile(path)
+		if err != nil {
+			t.Fatal(err)
+		}
+		fileMap[name] = false
+		dataMap[name] = data
+	}
+
+	// Add the public key file, and verify that
+	// server.pubkey contains the first 32 bytes from server.keys
+	// This test verifies this value directly.
+	const pkf = "server.pubkey"
+	fileMap[pkf] = false
+	path := filepath.Join(dir, "server.keys")
+	data, err := os.ReadFile(path)
+	if err != nil {
+		t.Fatal(err)
+	}
+	dataMap[pkf] = data[:32]
+
+	const rmf = "README"
+	fileMap[rmf] = false
+	dataMap[rmf] = []byte(ReadmeContents)
+
+	// Post the archive request
+	resp, err := http.Get(fmt.Sprintf("http://localhost:%v/api/v1/archive", gcas.httpPort))
+	if err != nil {
+		t.Fatal(err)
+	}
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		t.Fatal(fmt.Errorf("expected status 200, but got %d: %s", resp.StatusCode, string(body)))
+	}
+
+	reader := bytes.NewReader(body)
+
+	rzip, err := zip.NewReader(reader, int64(len(body)))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, zipf := range rzip.File {
+		// Make sure there is not an extra file in the zip contents,
+		// and keep track of the ones we find.
+		if _, ok := fileMap[zipf.Name]; ok {
+			fileMap[zipf.Name] = true
+		} else {
+			t.Fatal(fmt.Errorf("%v from zipfile is not a valid file", zipf.Name))
+		}
+
+		// Check each file against the original data.
+
+		f, err := zipf.Open()
+		if err != nil {
+			t.Fatal(fmt.Errorf("%v from zipfile could not be opened: %v", zipf.Name, err))
+		}
+		defer f.Close()
+
+		data, err := io.ReadAll(f)
+		if err != nil {
+			t.Fatal(err)
+		}
+
```

Comment:
these 3 newlines are all not needed, can just stack up the whole idea of open/read/check. Go is verbose enough without newlines that things are still quite readable - the way go handles the errors adds great visual signatures for each step of the process.

Reply:
I will go over the whole module to try to minimize newlines